pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -168,6 +207,160 @@ func newLighthouseAgent(cr *submarinerv1alpha1.ServiceDiscovery) *appsv1.Deploym
+	}
+	expectedCorefile := `supercluster.local:53 {
+lighthouse
+errors
+health
+ready
+}
+`
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      lighthouseCoreDNSName,
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Data: map[string]string{
+			"Corefile": expectedCorefile,
+		},
+	}
+}
+
+func newLigthhouseCoreDNSDeployment(cr *submarinerv1alpha1.ServiceDiscovery) *appsv1.Deployment {
+	replicas := int32(2)
+	labels := map[string]string{
+		"app":       lighthouseCoreDNSName,
+		"component": componentName,
+	}
+	matchLabels := map[string]string{
+		"app": lighthouseCoreDNSName,
+	}
+
+	terminationGracePeriodSeconds := int64(0)
+	defaultMode := int32(420)
+
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      lighthouseCoreDNSName,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: matchLabels,
+			},
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            lighthouseCoreDNSName,
+							Image:           getImagePath(cr, lighthouseCoreDNSImage),
+							ImagePullPolicy: "IfNotPresent",
+							Args: []string{
+								"-conf",
+								"/etc/coredns/Corefile",
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{Name: "config-volume", MountPath: "/etc/coredns", ReadOnly: true},
+							},
+						},
+					},
+
+					ServiceAccountName:            "submariner-operator",
+					TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
+					Volumes: []corev1.Volume{
+						{Name: "config-volume", VolumeSource: corev1.VolumeSource{ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{Name: lighthouseCoreDNSName},
+							Items: []corev1.KeyToPath{
+								{Key: "Corefile", Path: "Corefile"},
+							},
+							DefaultMode: &defaultMode,
+						}}},
+					},
+				},
+			},
+		},
+	}
+}
+
+func newLigthhouseCoreDNSService(cr *submarinerv1alpha1.ServiceDiscovery) *corev1.Service {
+	labels := map[string]string{
+		"app":       lighthouseCoreDNSName,
+		"component": componentName,
+	}
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      lighthouseCoreDNSName,
+			Labels:    labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Name:     "udp",
+				Protocol: "UDP",
+				Port:     53,
+				TargetPort: intstr.IntOrString{Type: intstr.Int,
+					IntVal: 53},
+			}},
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"app": lighthouseCoreDNSName,
+			},
+		},
+	}
+}
+
+func updateDNSConfigMap(client client.Client, k8sclientSet *clientset.Clientset, cr *submarinerv1alpha1.ServiceDiscovery) error {
+	configMaps := k8sclientSet.CoreV1().ConfigMaps("kube-system")
```

Comment:
how does it work on openshift that has 'dns-default' in namespace 'openshift-dns' ?

Reply:
I am working on the Openshift changes where it will update the cluster-dns-operator CR. I will raise a follow-up commit once it is ready

Reply:
ack.